src/app/Dashboard/AutomatedAnalysis/AutomatedAnalysisCard.tsx

```diff
@@ -0,0 +1,724 @@
+    generateReport,
+    handleStateErrors,
+  ]);
+
+  const handleErrorView = React.useCallback((): [string, undefined | (() => void)] => {
+    if (errorMessage === NO_RECORDINGS_MESSAGE) {
+      return ['', undefined];
+    } else if (errorMessage === RECORDING_FAILURE_MESSAGE) {
+      return ['Retry starting recording', startProfilingRecording];
+    } else if (errorMessage === FAILED_REPORT_MESSAGE) {
+      return ['Retry loading report', generateReport];
+    } else {
+      // errorMessage === INTERNAL_ERROR_MESSAGE
+      return ['Retry', generateReport];
+    }
+  }, [errorMessage, startProfilingRecording, generateReport]);
+
+  React.useEffect(() => {
+    addSubscription(
+      context.target.authFailure().subscribe(() => {
+        handleStateErrors('Authentication failure');
+      })
+    );
+  }, [addSubscription, context.target, handleStateErrors]);
+
+  React.useEffect(() => {
+    generateReport();
+  }, []);
+
+  React.useEffect(() => {
+    if (reportTime == 0 || !(usingArchivedReport || usingCachedReport)) {
+      return;
+    }
+    let interval, timerQuantity;
+    let now = Date.now();
+    const reportMillis = now - reportTime;
+    if (reportMillis < MINUTE_MILLIS) {
+      timerQuantity = Math.round(reportMillis / SECOND_MILLIS);
+      interval = SECOND_MILLIS - (reportMillis % SECOND_MILLIS);
+      setReportStalenessTimerUnits('second');
+    } else if (reportMillis < HOUR_MILLIS) {
+      timerQuantity = Math.round(reportMillis / MINUTE_MILLIS);
+      interval = MINUTE_MILLIS - (reportMillis % MINUTE_MILLIS);
+      setReportStalenessTimerUnits('minute');
+    } else if (reportMillis < DAY_MILLIS) {
+      timerQuantity = Math.round(reportMillis / HOUR_MILLIS);
+      interval = HOUR_MILLIS - (reportMillis % HOUR_MILLIS);
+      setReportStalenessTimerUnits('hour');
+    } else {
+      timerQuantity = Math.round(reportMillis / DAY_MILLIS);
+      interval = DAY_MILLIS - reportMillis * DAY_MILLIS;
+      setReportStalenessTimerUnits('day');
+    }
+    setReportStalenessTimer(timerQuantity);
+    const timer = setInterval(() => {
+      setReportStalenessTimer((reportStalenessTimer) => reportStalenessTimer + 1);
+    }, interval);
+    return () => clearInterval(timer);
+  }, [
+    setReportStalenessTimer,
+    setReportStalenessTimerUnits,
+    reportTime,
+    reportStalenessTimer,
+    usingArchivedReport,
+    usingCachedReport,
+  ]);
+
+  React.useEffect(() => {
+    setFilteredCategorizedEvaluation(
+      filterAutomatedAnalysis(
+        categorizedEvaluation,
+        targetAutomatedAnalysisFilters,
+        targetAutomatedAnalysisGlobalFilters,
+        showNAScores
+      )
+    );
+  }, [
+    categorizedEvaluation,
+    targetAutomatedAnalysisFilters,
+    targetAutomatedAnalysisGlobalFilters,
+    showNAScores,
+    filterAutomatedAnalysis,
+    setFilteredCategorizedEvaluation,
+  ]);
+
+  const onCardExpand = React.useCallback(() => {
+    setIsCardExpanded(!isCardExpanded);
```

Comment:
We should use `setIsCardExpanded((isCardExpanded) => !isCardExpanded);` to avoid using current state as state updates can be async.